### PR TITLE
[CVP-1452] Make CVP/operator-metadata-preparation should handle number in package file correctly

### PR DESCRIPTION
**Pull request by samvarankashyap**
Fixes CVP-1452 partly
Working on another fix to manage outputs of playbooks clearly. 

Regardless of versioning make operator-metadata-preparation should handle number in the package file correctly

### CODE REVIEW

File: roles/parse_operator_metadata/tasks/main.yml [hunk 1]
```diff
@@ -71,12 +71,14 @@
       current_channel: "{{ package_vars['channels'][0]['name'] }}"
     when: current_channel == ""
 
-  - name: "Determine the current clustersourceversion for the operator channel"
```
Comment (wheelerlaw) [inline roles/parse_operator_metadata/tasks/main.yml:74]: This was correct, why did you change it?

**Comment by wheelerlaw** [comment]
Do you have an example of it running?

File: roles/parse_operator_metadata/tasks/main.yml [hunk 2]
```diff
@@ -71,12 +71,14 @@
...
+  - name: "Determine the current clustersourceversion for operator channel"
     set_fact:
-      current_csv : "{{ package_vars['channels'] | json_query(query) }}"
-    vars:
-      query: "[?name=='{{ current_channel }}'].currentCSV"
-    failed_when: current_csv | length == 0
+      current_csv: "{{ package_vars['channels'] | determine_clusterservice_version(current_channel) }}"
+
+  - name: "Fail when current clusterSourceVersion is null"
```
Comment (dirgim) [inline roles/parse_operator_metadata/tasks/main.yml:78]: nit: ClusterServiceVersion is the correct name

**Review by dirgim — APPROVED** [review]
LGTM

**Comment by samvarankashyap** [comment]
@wheelerlaw Any more comments?
